.github/workflows/docker-build-and-run.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and Run Docker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
It's set to run for all pull requests. The docker build was very fast, but the run part won't be. But I'm also thinking not every pull request to main needs to build the docker image. Maybe just the main branch itself? Or, like the analysis pipeline, the branches that start with `docker/**` could also build it?